src/OpenIddict.Server.Quartz/OpenIddictServerQuartzJob.cs

```diff
@@ -0,0 +1,154 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/openiddict/openiddict-core for more information concerning
+ * the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OpenIddict.Abstractions;
+using Quartz;
+using static OpenIddict.Abstractions.OpenIddictExceptions;
+using SR = OpenIddict.Abstractions.OpenIddictResources;
+
+namespace OpenIddict.Server.Quartz
+{
+    /// <summary>
+    /// Represents a Quartz.NET job performing scheduled tasks for the OpenIddict server feature.
+    /// </summary>
+    [DisallowConcurrentExecution]
+    public class OpenIddictServerQuartzJob : IJob
+    {
+        private readonly ILogger _logger;
+        private readonly IOptionsMonitor<OpenIddictServerQuartzOptions> _options;
+        private readonly IServiceProvider _provider;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="OpenIddictServerQuartzJob"/> class.
+        /// </summary>
+        public OpenIddictServerQuartzJob() => throw new InvalidOperationException(SR.GetResourceString(SR.ID1081));
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="OpenIddictServerQuartzJob"/> class.
+        /// </summary>
+        /// <param name="logger">The logger.</param>
+        /// <param name="options">The OpenIddict server Quartz.NET options.</param>
+        /// <param name="provider">The service provider.</param>
+        public OpenIddictServerQuartzJob(
+            ILogger<OpenIddictServerQuartzJob> logger,
+            IOptionsMonitor<OpenIddictServerQuartzOptions> options,
+            IServiceProvider provider)
+        {
+            _logger = logger;
+            _options = options;
+            _provider = provider;
+        }
+
+        /// <summary>
+        /// Gets the default identity assigned to this job.
+        /// </summary>
+        public static JobKey Identity { get; } = new JobKey(
+            name: typeof(OpenIddictServerQuartzJob).Name,
+            group: typeof(OpenIddictServerQuartzJob).Assembly.GetName().Name!);
+
+        /// <inheritdoc/>
+        public async Task Execute(IJobExecutionContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            // TODO: determine whether errors should be swallowed or should be re-thrown to be logged by Quartz.NET.
+
+            // Note: this job is registered as a transient service. As such, it cannot directly depend on scoped services
+            // like the core managers. To work around this limitation, a scope is manually created for each invocation.
+            var scope = _provider.CreateScope();
```

Comment:
There's also `MicrosoftDependencyInjectionScopedJobFactory` which should provide scoping.

Reply:
`MicrosoftDependencyInjectionScopedJobFactory` is a nice option, but it's hard for a lib to assume it's the option chosen by the developer (you'll generally get a cryptic error message from DI when trying to resolve a scoped service from a transient).

We could probably find a way to force users to use `MicrosoftDependencyInjectionScopedJobFactory` instead of `MicrosoftDependencyInjectionJobFactory`, but it would affect every other job (not just ours), so I'm not sure it would be a nice thing to do in a library 😕